joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/AstCreatorBase.scala

```diff
@@ -112,8 +113,21 @@ abstract class AstCreatorBase(filename: String) {
   /** For a given call node, arguments, and optionally, a receiver, create an AST that represents the call site. The
     * main purpose of this method is to automatically assign the correct argument indices.
     */
-  def callAst(callNode: NewCall, arguments: Seq[Ast] = List(), receiver: Option[Ast] = None): Ast = {
+  def callAst(
+    callNode: NewCall,
+    arguments: Seq[Ast] = List(),
+    receiver: Option[Ast] = None,
+    withRecvArgEdge: Boolean = false
+  ): Ast = {
+    astWithRecvAndArgs(callNode, arguments, receiver, withRecvArgEdge)
+  }
 
+  def astWithRecvAndArgs(
+    rootNode: NewNode,
+    arguments: Seq[Ast] = List(),
+    receiver: Option[Ast] = None,
+    withRecvArgEdge: Boolean = false
+  ): Ast = {
```

Comment:
We dont need the same functionality under 2 names. Please just move the body of this function in `callAst`.